src/plugins/recursion/README.md

```diff
@@ -81,10 +81,186 @@ kdb set /recursive/define/#2/ref/#1 /recursive/define/#5
+is inspired from libreoffice and simplified.
+
+Imagine an editor with a `File` menu with a `Print` option as well as a submenu called `Settings`.
+The `Settings` submenu has `Global Settings` and  `Autocorrect Options`
+Beside the `File` menu we also have a `Tool` menu which also has the same `Settings` submenu but also
+has gotten a `Extension Manager` and a Menu `Macros`. Macros has again a submenu `Organize Macros` which
+again is a menu and has `Macro Basics`.
+
+The following picture illustrates the menu tree:
+![menu-example](example.png?raw=true "Example")
+
+So how can we map this into an elektra configuration?
+
+First we create a config and mount it. Next we will set the specification for all menu entries as following:
+```
+kdb mount config.dump /editor dump recursion
+sudo kdb setmeta /editor check/recursion/vertex "menu"
+sudo kdb setmeta /editor check/recursion/ref "menuref"
+```
+Each key definition in the form of `/editor/menu` will now be treated as recursive entry. You can define arbitrary
+many menues like this: `/editor/menu/#[d]` where `[d]` stands for a positive number.
+
+Menues can be linked via `menuref` like this: `/editor/menu/#[d]/menuref/#[d]` and is again in the array notation.
+Note that `menu` and `menuref` are our defined keywords in the setmeta command from above.
+
+Next lets create all distinct menu entries such as `File`, `Print` etc. Note that the Settings
+ menu appears twice but we will only need to declare it once. The order does not matter:
+
+```
+kdb set /editor/menu/#0 File
+kdb set /editor/menu/#1 Tools
+kdb set /editor/menu/#2 Print
+kdb set /editor/menu/#3 Settings
+kdb set /editor/menu/#4 "Global Settings"
+kdb set /editor/menu/#5 "Autocorrect Options"
+kdb set /editor/menu/#6 "Extension Manager"
+kdb set /editor/menu/#7 "Macros"
+kdb set /editor/menu/#8 "Organize Macros"
+kdb set /editor/menu/#9 "Macro Basics"
+```
+
+Now that we have defined we can link together all menues. Lets start with the `File` menu:
+
+```
+kdb set /editor/menu/#0/menuref/#0 /editor/menu/#2
+kdb set /editor/menu/#0/menuref/#1 /editor/menu/#3
+```
+
+Next we link the `Settings` menu:
+
+```
+kdb set /editor/menu/#3/menuref/#0 /editor/menu/#5
+kdb set /editor/menu/#3/menuref/#1 /editor/menu/#6
+```
+
+Now we link the `Tools` section:
+```
+kdb set /editor/menu/#1/menuref/#0 /editor/menu/#5
+kdb set /editor/menu/#1/menuref/#1 /editor/menu/#3
+kdb set /editor/menu/#1/menuref/#2 /editor/menu/#7
+```
+
+Note that `Settings` is already linked. We do not need to do it for the `Tools` menu again.
+
+Finally we link all submenus of `Macros`:
+
+```
+kdb set /editor/menu/#7/menuref/#0 /editor/menu/#8
+kdb set /editor/menu/#8/menuref/#0 /editor/menu/#9
+```
+
+### Example of wrong settings
+
+In the above example the following settings will be rejected:
+
+```
+#Link "Macro Basics" to the parent menu "Macros" and create an endless loop
+kdb set /editor/menu/#9/menuref/#0 /editor/menu/#7
+ERR: 198
```

Comment:
Please use shell recorder syntax

Reply:
Do you have a link to a tutorial for that?

Reply:
The following documents should be helpful:

- [ReadMe of the Markdown Shell Recorder (MSR)](https://master.libelektra.org/tests/shell/shell_recorder/tutorial_wrapper/README.md)
- [the MSR syntax check](https://master.libelektra.org/tests/shell/shell_recorder/tutorial_wrapper/SyntaxCheck.md)

. Many of the plugins also contain some basic MSR tests. For example, the ReadMe of [YAML CPP](https://www.libelektra.org/plugins/yamlcpp) contains a few example tests.